File: backend/src/main/java/br/ufpr/sept/so2/modules/presenca/api/dto/EventoResponse.java
package br.ufpr.sept.so2.modules.presenca.api.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;
import java.util.Map;
import java.util.UUID;

public record EventoResponse(
        UUID id,
        UUID idAnfitriao,
        String titulo,
        OffsetDateTime inicioEm,
        OffsetDateTime fimEm,
        int cargaHoraria,
        String attendanceMode,
        String estado,
        String situacaoPresenca,
        boolean janelaAtiva,
        @JsonProperty("_links") Map<String, String> links
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/presenca/api/dto/EventoResponse.kt
package br.ufpr.sept.so2.modules.presenca.api.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.OffsetDateTime
import java.util.UUID

data class EventoResponse(
    val id: UUID,
    val idAnfitriao: UUID?,
    val titulo: String,
    val inicioEm: OffsetDateTime,
    val fimEm: OffsetDateTime,
    val cargaHoraria: Int,
    val attendanceMode: String,
    val estado: String,
    val situacaoPresenca: String,
    val janelaAtiva: Boolean,
    @get:JsonProperty("_links")
    val links: Map<String, String>,
)
